src/main/java/org/takes/facets/auth/signatures/SiHmac.java

```diff
@@ -0,0 +1,180 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.facets.auth.signatures;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Formatter;
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import lombok.EqualsAndHashCode;
+
+/**
+ * HMAC codec which supports 256, 384 and 512 bit hash.
+ *
+ * <p>
+ * The class is immutable and thread-safe.
+ *
+ * @author Sven Windisch (sven.windisch@gmail.com)
+ * @version $Id$
+ * @since 1.4
+ */
+@EqualsAndHashCode
+public final class SiHmac implements Signature {
+    /**
+     * The HMAC 256 bit variant.
+     */
+    public static final int HMAC256 = 256;
+
+    /**
+     * The HMAC 384 bit variant.
+     */
+    public static final int HMAC384 = 384;
+
+    /**
+     * The HMAC 512 bit variant.
+     */
+    public static final int HMAC512 = 512;
+
+    /**
+     * The encryption key.
+     */
+    private final byte[] key;
+
+    /**
+     * The bit length parameter.
+     */
+    private final int bits;
+
+    /**
+     * Ctor.
+     *
+     * @param key
+     *  The encryption key
+     */
+    public SiHmac(final String key) {
+        this(key.getBytes(Charset.defaultCharset()), SiHmac.HMAC256);
+    }
+
+    /**
+     * Ctor.
+     *
+     * @param key
+     *  The encryption key
+     * @param bits
+     *  The signature bit length
+     */
+    public SiHmac(final String key, final int bits) {
+        this(key.getBytes(Charset.defaultCharset()), bits);
+    }
+
+    /**
+     * Ctor.
+     *
+     * @param key
+     *  The encryption key
+     * @param bits
+     *  The signature bit length
+     */
+    public SiHmac(final byte[] key, final int bits) {
+        this.key = key.clone();
+        this.bits = SiHmac.bitLength(bits);
+    }
+
+    @Override
+    public byte[] sign(final byte[] data) throws IOException {
+        return this.encrypt(data);
+    }
+
+    /**
+     * Signature bit length.
+     *
+     * @return The bitlength
+     */
+    public int bitlength() {
+        return this.bits;
+    }
+
+    /**
+     * Check for correct bit length.
+     *
+     * @param bits
+     *  Given bit length
+     * @return Original bit length when appropriate, 256 bits otherwise.
+     */
+    private static int bitLength(final int bits) {
+        int correct = bits;
+        if (bits != SiHmac.HMAC256
+            && bits != SiHmac.HMAC384
+            && bits != SiHmac.HMAC512) {
+            correct = SiHmac.HMAC256;
+        }
+        return correct;
+    }
+
+    /**
+     * Encrypt the given bytes using HMAC.
+     *
+     * @param bytes
+     *  Bytes to encrypt
+     * @return Encrypted bytes
+     * @throws IOException
+     *  for all unexpected exceptions
+     */
+    @SuppressWarnings("resource")
```

Comment:
@yegor256 This is kinda silly. Constructing the Formatter without parameters forces it to use an internal StringBuilder which is totally OK for our use. But that StringBuilder is not closable, so calling close() on the Formatter is useless in terms of saving resources. Eclipse doesn't know this and warns anyway. That's why I suppressed it. I'll close it, if you like it better that way.